### ISSUE TITLE
AWS EKS pilotCertProvider gives remote error: tls: bad certificate

### ISSUE THREAD

**Issue by christian-posta** [istio/istio#24997]
**Bug description**
Istiod doesn't start correctly, and no workload sidecar proxies can connect.
See this repeatedly in the logs:

```
2020-06-24T15:53:14.018549Z    info    validationController    Reconcile(enter): retry dry-run creation of invalid config
2020-06-24T15:53:14.023959Z    info    http: TLS handshake error from 192.168.147.176:59130: remote error: tls: bad certificate
2020-06-24T15:53:14.026920Z    info    validationController    Not ready to switch validation to fail-closed: dummy invalid config not rejected
2020-06-24T15:53:14.027244Z    info    validationController    validatingwebhookconfiguration istiod-istio-system (failurePolicy=Ignore,
```

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Steps to reproduce the bug**
When installing Istio using pilotCertProvider == Kubernetes on AWS EKS (< kube 1.16) we run into this issue.

**How was Istio installed?**
Istio operator

**Environment where bug was observed (cloud vendor, OS, etc)**
AWS EKS

I'm adding this here after the fact, just so it's included in the GitHub issues and other folks can find it if they run into it.

Basically when using pilotCertProvider of Kubernetes, AWS EKS has a bug... when we try to get the DNS certificates for istiod.istio-system.svc using Chiron, AWS returns a cert WITHOUT the SAN corresponding to the DNS hosts that were requested. This causes the validation web hook to fail (or any web hooks) to call back to Istiod. 

This bug on AWS EKS (https://github.com/awslabs/amazon-eks-ami/issues/341) seems to have been fixed in more recent versions.

**Comment by christian-posta** [comment]
Using a version of EKS Kubernetes (1.16+), you should no longer see this AWS bug
